spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -916,7 +916,10 @@ object CometSparkSessionExtensions extends Logging {
   private[comet] def isCometShuffleEnabled(conf: SQLConf): Boolean =
     COMET_EXEC_SHUFFLE_ENABLED.get(conf) &&
       (conf.contains("spark.shuffle.manager") && conf.getConfString("spark.shuffle.manager") ==
-        "org.apache.spark.sql.comet.execution.shuffle.CometShuffleManager")
+        "org.apache.spark.sql.comet.execution.shuffle.CometShuffleManager") &&
+      // TODO: AQE coalesce partitions feature causes Comet shuffle memory leak.
```

Comment:
As I mentioned in the ticket, when coalesce partitions is enabled, Spark will combine the partitions of multiple reducers. I suspect that causes incorrect format to read for Java Arrow StreamReader.

We should address this issue further to unblock Comet shuffle with coalesce partitions. But for now, I think it is better to disable it temporarily for the cases we know it will cause some issues.

Reply:
+1 for putting this behind a config and disabling by default until we have a solution

Reply:
Is there an issue logged for the followup?

Reply:
> Is there an issue logged for the followup?

Let me create one.

Reply:
Created #387 to track it.

Reply:
Thanks @advancedxy. I debugged this issue but didn't find a quick fix so decided to disable it temporarily.

I took a look at your branch. The Java allocator instance will report the memory leak when getting closed if it has allocated memory size is larger then zero. So as you did, if we find there is non-zero number (`getAllocatedMemory` > 0), we don't close the allocator, it won't report that.

However, I'm not sure if it is correct fix and if we will ignore real memory leak. Maybe it is a false positive one. But If it is real memory leak and we ignore it, it will be a potential issue.